lib/scala/kg/src/main/scala/formats/CsvReader.scala

```diff
@@ -0,0 +1,51 @@
+package formats
+
+import com.github.tototoshi.csv.CSVReader
+import java.nio.file.Path
+import java.io.BufferedInputStream
+import java.io.FileInputStream
+import java.io.InputStream
+import java.io.InputStreamReader
+import org.apache.commons.compress.compressors.CompressorStreamFactory
+import org.apache.commons.compress.compressors.CompressorException
+import java.io.Reader
+import com.github.tototoshi.csv.CSVFormat
+import org.apache.commons.lang3.StringUtils
+
+abstract class CsvReader[T](protected val csvReader: CSVReader) extends AutoCloseable with Iterable[T] {
+  override def close(): Unit =
+    csvReader.close()
+
+  protected implicit class RowWrapper(row: Map[String, String]) {
+    def getNonBlank(key: String) =
+      row.get(key).flatMap(value => if (!StringUtils.isBlank(value)) Some(value) else None)
+
+    def getList(key: String, delim: String) =
+      getNonBlank(key).flatMap(value => Option(value.split(delim).toList)) getOrElse List[String]()
+  }
+
+  def iterator: Iterator[T]
+}
+
+object CsvReader {
+  def openCsvReader(filePath: Path, csvFormat: CSVFormat): CSVReader = {
```

Comment:
I think there's probably a more Scala-esque way to do this e.g., provide a factory method for subclasses, similar to Python @classmethod. But I don't know what it is.